Sources/NIO/UniversalBootstrapSupport.swift

```diff
@@ -167,6 +178,58 @@ public struct NIOClientTCPBootstrap {
         return NIOClientTCPBootstrap(self.tlsEnablerTypeErased(self.underlyingBootstrap),
                                      tlsEnabler: self.tlsEnablerTypeErased)
     }
+    
+    /// A channel option which can be applied to bootstrap using shorthand notation.
+    /// - See: NIOClientTCPBootstrap.channelOptions(_ options: [Option])
+    public struct Option {
+        private let data: ShorthandOption
+        
+        private init(_ data: ShorthandOption) {
+            self.data = data
+        }
+        
+        /// Apply the contained option to the supplied NIOClientTCPBootstrap
+        /// - Parameter to: bootstrap to apply this option to.
+        /// - Returns: the modified bootstrap (currently the same one mutated)
+        @usableFromInline
+        func applyOption(to clientBootstrap: NIOClientTCPBootstrap) {
+            data.applyOption(to: clientBootstrap)
+        }
+        
+        fileprivate enum ShorthandOption {
+            case reuseAddr
+            case allowRemoteHalfClosure(Bool)
+            
+            func applyOption(to bootstrap: NIOClientTCPBootstrap) {
+                switch self {
+                case .reuseAddr:
+                    _ = bootstrap.channelOption(ChannelOptions.socketOption(.reuseaddr), value: 1)
+                case .allowRemoteHalfClosure(let value):
+                    _ = bootstrap.channelOption(ChannelOptions.allowRemoteHalfClosure, value: value)
+                }
+            }
+        }
+    }
+}
+
+// Hashable for the convenience of users.
+extension NIOClientTCPBootstrap.Option: Hashable {}
+extension NIOClientTCPBootstrap.Option.ShorthandOption: Hashable {}
+
+/// Approved shorthand tcp client options.
+extension NIOClientTCPBootstrap.Option {
+    /// Option to reuse address.
+    /// - See:  NIOBSDSocket.Option.reuseaddr
+    public static let allowImmediateEndpointAddressReuse = NIOClientTCPBootstrap.Option(.reuseAddr)
+    
+    /// - See: `AllowRemoteHalfClosureOption`.
+    public static let allowRemoteHalfClosure =
```

Comment:
this will work, but we now have two copies of `*Bootstrap.Option`. Should we maybe unify those?

Reply:
Did I address your concerns with NIOTCPShorthandOption being shared between the various TCP data channel bootstraps?  There are still the other options for listen, pipe, udp etc which I think are probably fine to be separate as they are a different thing which just happens to have the same name.